core/web-assets/src/main/assets/js/lib/onms-schedule-editor/templates/schedule-editor.tpl.html

```diff
@@ -0,0 +1,169 @@
+
+    <!-- Days per Month -->
+    <div class="" ng-if="options.type === 'daysPerMonth'">
+        <!-- Other options -->
+        <div class="form-group">
+            <label>At</label>
+            <onms-time-input
+                    ng-model="options.at"
+                    options="options.at"
+            />
+        </div>
+
+        <div class="">
+            <label>On</label>
+
+            <div class="form-inline my-1 ml-1">
+                <input class="" type="radio" ng-model="options.dayOfMonthToggle" value="dayOfMonth">
+                <span class="mx-1">the</span>
+                <select class="form-control custom-select" ng-model="options.dayOfMonth">
+                    <option value="1">1st</option>
+                    <option value="2">2nd</option>
+                    <option value="3">3rd</option>
+                    <option value="4">4th</option>
+                    <option value="5">5th</option>
+                    <option value="6">6th</option>
+                    <option value="7">7th</option>
+                    <option value="8">8th</option>
+                    <option value="9">9th</option>
+                    <option value="10">10th</option>
+                    <option value="11">11th</option>
+                    <option value="12">12th</option>
+                    <option value="13">13th</option>
+                    <option value="14">14th</option>
+                    <option value="15">15th</option>
+                    <option value="16">16th</option>
+                    <option value="17">17th</option>
+                    <option value="18">18th</option>
+                    <option value="19">19th</option>
+                    <option value="20">20th</option>
+                    <option value="21">21st</option>
+                    <option value="22">22nd</option>
+                    <option value="23">23rd</option>
+                    <option value="24">24th</option>
+                    <option value="25">25th</option>
+                    <option value="26">26th</option>
+                    <option value="27">27th</option>
+                    <option value="28">28th</option>
+                    <option value="29">29th</option>
+                    <option value="30">30th</option>
+                    <option value="31">31st</option>
+                    <option value="L">Last</option>
+                </select>
+
+                <span class="ml-1">day of every month</span>
+            </div>
+
+            <!-- Short selection -->
+            <div class="form-inline my-1 ml-1">
+                <input class="" type="radio" ng-model="options.dayOfMonthToggle" value="dayOfWeek">
+                <span class="mx-1">the</span>
+                <select class="form-control custom-select" ng-model="options.weekOfMonth">
+                    <option value="1">first</option>
+                    <option value="2">second</option>
+                    <option value="3">third</option>
```

Comment:
This year August has 5 weeks, should there be a `fourth`? I realize it's quite an edge case, but...

Reply:
Technically you can define a `fourth` or `fifth`.
However if you define, let's say the `fifth Thursday` and there is no `fifth Thursday` in that month, the report is not delivered.
From my point of view the `Schedule Editor` should make creating a schedule very easy.
If you can configure an edge case it may not be very trivial what actually happens behind the scene.
Therefore I decided to drop the `fourth` and `fifth`, just use `Last` in that scenario.
If required I can add a fourth, but would leave the fifth. 
Besides that you can always use `Cron Expression` and define everything in more detail.

> Note that if you specify “#5” and there is not 5 of the given day-of-week in the month, then no firing will occur that month.
Source: http://www.quartz-scheduler.org/documentation/quartz-2.3.0/tutorials/crontrigger.html